app/src/main/java/com/example/triptracker/view/HomeScreen.kt

```diff
@@ -162,19 +205,131 @@ fun convertPinListToString(pinList: List<String>): String {
   }
 }
 
+/**
+ * Displays the search bar at the top of the screen
+ *
+ * @param onBackClicked: Function to call when the back button is clicked
+ * @param onSearchActivated: Function to call when the search bar is activated
+ * @param viewModel: HomeViewModel to use for searching itineraries
+ * @param isNoResultFound: Boolean to indicate if no results were found
+ * @param navigation: Navigation object to use for navigation, i.e when an itinerary is clicked
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchBar(
```

Comment:
It would be beneficial to have distinct names for the two search-related composable functions. Currently, both composable functions share the name “SearchBar”, which might lead to confusion regarding their purposes.